source/lib/omnitrace/library/roctracer.cpp

```diff
@@ -629,6 +647,23 @@ hip_api_callback(uint32_t domain, uint32_t cid, const void* callback_data, void*
         default: break;
     }
 
+    if(get_perfetto_roctracer_per_stream())
+    {
+        static thread_local std::unordered_set<uintptr_t> seen_queues;
```

Comment:
I am not certain you want thread locality here. I am not sure what perfetto's behavior is when you create tracks with the same UUID on different threads -- does it reference the same track or are they different tracks? I think you might want submissions to the same queue from different threads to be in the same track